homeassistant/components/climate/ambiclimate.py

```diff
@@ -0,0 +1,271 @@
+"""Support for Ambiclimate ac."""
+
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.climate import ClimateDevice, PLATFORM_SCHEMA
+from homeassistant.components.climate.const import (
+    ATTR_CURRENT_HUMIDITY, DOMAIN, SUPPORT_TARGET_TEMPERATURE,
+    SUPPORT_ON_OFF, STATE_HEAT)
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.const import (ATTR_NAME, ATTR_TEMPERATURE,
+                                 STATE_OFF, TEMP_CELSIUS)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+REQUIREMENTS = ['ambiclimate==0.1.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+AMBICLIMATE_DATA = 'ambiclimate'
+ATTR_VALUE = 'value'
+CONF_CLIENT_ID = 'client_id'
+CONF_CLIENT_SECRET = 'client_secret'
+CONFIGURATOR_DESCRIPTION = 'To link your Ambiclimate account, ' \
+                           'click the link, login, and authorize:'
+CONFIGURATOR_LINK_NAME = 'Link Ambiclimate account'
+CONFIGURATOR_SUBMIT_CAPTION = 'I authorized successfully'
+DEFAULT_NAME = 'Ambiclimate'
+SERVICE_COMFORT_FEEDBACK = 'send_comfort_feedback'
+SERVICE_COMFORT_MODE = 'set_comfort_mode'
+SERVICE_TEMPERATURE_MODE = 'set_temperature_mode'
+STORAGE_KEY = 'ambiclimate_auth'
+STORAGE_VERSION = 1
+
+AUTH_CALLBACK_NAME = 'api:ambiclimate'
+AUTH_CALLBACK_PATH = '/api/ambiclimate'
+
+SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE |
+                 SUPPORT_ON_OFF)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
```

Comment:
We don't want users entering oauth client/id secrets into `configuration.yaml` anymore.  The recommendation is to capture this during a config flow.